frontend/src/components/errors/NotFoundPage.tsx

```diff
@@ -0,0 +1,49 @@
+import { useTranslation } from 'next-i18next'
+import { useRouter } from 'next/router'
+
+import { Button, Grid, Typography } from '@material-ui/core'
+
+import { createStyles, Theme, makeStyles } from '@material-ui/core/styles'
+import ArrowBackIcon from '@material-ui/icons/ArrowBack'
+
+import NotFoundIllustration from './assets/NotFoundIllustration'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      flexGrow: 1,
+      height: '100vh',
+      margin: 0,
+      backgroundColor: theme.palette.secondary.light,
+    },
+  }),
+)
+
+export default function NotFoundPage() {
+  const { t } = useTranslation()
+  const router = useRouter()
+
+  const classes = useStyles()
+
+  return (
+    <Grid container className={classes.root}>
```

Comment:
Maybe we can introduce the header navigation while we're showing the 404 page

Reply:
Yup, we can do that!

What do you think only the header navigation or the whole layout component (header + footer)?

Reply:
There's a catch with the whole layout  - sometimes when it depends on dynamic component it cannot be rendered during Static Site Generation (SSG) phase and `_error.tsx` is always rendered during that phase.
Anyway I don't we have any dynamic elements in the layout so far, so it wont be a problem to render it in the error page. 

You might try it and see if it works with no extra effort.

Reply:
Yeah, works fine with the current state of the layout.

If we include the error pages with the Layout maybe we need to leave somewhere note to mention that introducing dynamic component will break the error pages?

![image](https://user-images.githubusercontent.com/37806520/124086032-551f6f00-da59-11eb-858f-751f6665b778.png)

Reply:
Looks awesome! Let's keep it like that.

Don't worry about future updates if they break the app it will be shown during PR checks as we run `next build` during that phase